cli/cli.py

```diff
@@ -523,6 +608,18 @@ def get_option_type_info() -> dict[str, Any]:
     return option_type_info
 
 
+def process_tables_options(opts: dict[str, Any], algo_name: str) -> dict[str, Any]:
+    if algo_name not in TASK_INFO[Task.ind].algos:
+        return opts
+
+    for option_name, parse_func in ((TABLES_LIST, parse_tables_list_file),
+            (TABLES_DIRECTORY, parse_tables_directory)):
+        opts[TABLES] = list(opts[TABLES]) + parse_func(opts[option_name])
+        opts.pop(option_name)
```

Comment:
```suggestion
        value = opts.pop(option_name)
        opts[TABLES] = list(opts[TABLES]) + parse_func(value)
```
`dict.pop` [returns the value](https://docs.python.org/3.8/library/stdtypes.html#dict.pop). If the value is not needed (which is *not* the case here), `del dict[key]` should be used.